### PR TITLE
Simple admin interface for eZ Comments

### PR DESCRIPTION
**Pull request by ghost**
A simple admin interface for comments, requested by one of our clients so sharing it upstream.

Admin interface lists all the available comments sorted by modified time, with pager at the bottom. Allows editing of comments and multiple delete.

### CODE REVIEW

**Comment by bdunogier** [comment]
shouldn't this be a pull request to https://github.com/ezsystems/ezcomments instead ? It would make more sense here.

**Comment by xc** [comment]
Cool feature. Thank you.
I will take a look soon.

**Comment by netgen** [comment]
@bdunogier Well, it is, isn't it :)

"netgen submitted to ezsystems/ezcomments 15 minutes ago"

**Comment by bdunogier** [comment]
haha. I blame myself :-)

I haven't looked in depth, but it looks promising ! Any screenshot for us ? :)

**Comment by netgen** [comment]
Hahah :)

Here's the screen: http://i55.tinypic.com/4jx641.png

**Comment by xc** [comment]
Haven't look into code. But from the screenshot, I'm thinking if it's suitable to create ezcomments tab for it since it's only lising comments (I was thinking there might be more features if we need a tab)

How about adding a link to left menu in setup tab (like workflow,etc)?

**Comment by bdunogier** [comment]
I don't think it really is a setup entry. Maybe a user generated content one tab ?

**Comment by xc** [comment]
How about putting comment list into dashboard tab? If we developed more complex features like moderation, subscription managing, etc, we can have a tab (named ugc/comments/whatever).

**Comment by netgen** [comment]
We added a tab simply to allow later feature additions to be easier. This way ezcomments would be self contained and not scattered throughout the admin interface. Also, it was important for us to have one place where we would administer ALL created comments without need for going through content to find them.

EDIT: Look as eZ Find for example, also has a tab with only one feature available (elevation)

**Comment by bdunogier** [comment]
A dashboard widget could be an interesting feature for later. We should provide a clear API for these first :-)

File: design/admin2/templates/comment/list.tpl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                <tr>
+                    {if $remove_comments_allowed}<th class="tight"><img src={'toggle-button-16x16.gif'|ezimage} alt="{'Select all comments'|i18n('ezcomments/comment/list')}" onclick="ezjs_toggleCheckboxes( document.commentlist, 'DeleteIDArray[]' ); return false;" /></th>{/if}
+                    <th>{'Comment'|i18n('ezcomments/comment/list')}</th>
+                    <th>{'Node'|i18n('ezcomments/comment/list')}</th>
+                    <th>{'Modified'|i18n('ezcomments/comment/list')}</th>
+                    <th style="text-align:right">{'IP address'|i18n('ezcomments/comment/list')}</th>
+                    {if $edit_allowed}<th class="tight">&nbsp;</th>{/if}
+                </tr>
+
+                {def $comments_count = fetch(comment, comment_list_by_content_list)|count}
```
Comment (xc) [inline design/admin2/templates/comment/list.tpl]: I'm a bit worry about the performance here. Fetching all comments once might not be a good idea...
 Maybe we should have a special fetch function for counting?

**Comment by netgen** [comment]
By all means. I can create something like fetch(comment, comment_count_by_content_list)

**Comment by xc** [comment]
Maybe we can reuse the function comment_count, with contentobject_id=false parameter?

**Comment by netgen** [comment]
Why not, seems to be working fine. I can push the commit using comment_count.

**Comment by xc** [comment]
I think I need to change some for that function..

ezcomcomment.php:
function countByContent:
...
        if( isset( $contentObjectID ) )
        {
            $cond['contentobject_id'] = $contentObjectID;
        }
        if ( isset( $languageID ) )
        {
            $cond['language_id'] = $languageID;
        }
        if ( isset( $status ) )
...

comment_count: 
...
                                                   array( 'name'=> 'contentobject_id',
                                                          'type'=>'integer',
                                                          'required'=>false,
                                                          'default'=>null
                                                    ),
                                                   array( 'name'=> 'language_id',
                                                          'type'=> 'integer',
                                                          'required' => false,
                                                          'default'  => null
                                                   ),
...

**Comment by netgen** [comment]
Yeah, I did that already :) It would be inside of a commit :)

**Comment by netgen** [comment]
Closing this pull request because it was made from my master branch, which is wrong. Correct pull request coming up.

**Pull request closed without merging**